src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    num: 0,
+
+    add(value) {
+      this.num += value;
+    },
+
+    subtract(value) {
+      this.num -= value;
+    },
+
+    divide(value) {
+      this.num /= value;
+    },
+
+    multiply(value) {
+      this.num *= value;
+    },
+
+    reset() {
+      this.num = 0;
+
+      return this;
+    },
+
+    get result() {
+      return this.num;
+    },
+
+    operate(callback, value) {
+      const func = callback.bind(this, value);
```

Comment:
just 

```
            callback.call(this, value);
            return this;
```

Reply:
the better approach I saw in Artem decision latter. 

callback(this.num, value);

Reply:
Thank you!)